lib/AST/ASTPrinter.cpp

```diff
@@ -2610,7 +2636,25 @@ void PrintAST::printOneParameter(const ParamDecl *param, bool Curried,
   // Set and restore in-parameter-position printing of types
   {
     llvm::SaveAndRestore<bool> savePrintParam(Options.PrintAsInParamType, true);
+    // FIXME: don't do if will be using type repr printing
+    printParameterFlags(Printer, Options, paramFlags);
```

Comment:
This whole area actually needs to be more complicated, and we're not exhaustively testing nested nested closures while mixing escapability. I'm testing that now.